docs/tutorials/hamiltonians.md

```diff
@@ -19,6 +19,19 @@ hamilt = hamiltonian_factory(n_qubits, interaction=Interaction.ZZ)
 print(hamilt) # markdown-exec: hide
 ```
 
+Alternatively, a custom interaction function can also be defined. The input should be two integer indices $i$ and $j$ and it should return a composition of pauli terms representing the interaction between qubits $i$ and $j$:
+
+```python exec="on" source="material-block" result="json" session="hamiltonians"
+def custom_int(i: int, j: int):
```

Comment:
Would it make sense to pass a pair object ? Ie. a couple of ints ? I guess this is what you get out of the register when you extract the complete graph (which sounds to be the most common use case but I might be wrong there).

Reply:
The signature here follows the other interaction functions that are pre-defined for use in the `hamiltonian_factory`. To change it here I would also have to change those (which I could do).

Either way, the current signature can be easily used with the information the register provides like so:
```
from qadence import *

reg = Register.line(5)

def custom_int(i: int, j: int):
    return X(i) @ X(j) + Y(i) @ Y(j)

add(custom_int(*p) for p in reg.all_node_pairs)
```

Reply:
Also doable indeed.